pgjdbc/src/main/java/org/postgresql/jdbc/BatchResultHandler.java

```diff
@@ -152,8 +152,10 @@ public void handleError(SQLException newError) {
 
       String queryString = "<unknown>";
       if (resultIndex < queries.length) {
-        queryString = queries[resultIndex].toString(
-            parameterLists == null ? null : parameterLists[resultIndex]);
+    	if(pgStatement.getPGConnection().getLogServerErrorDetail()) {
+	        queryString = queries[resultIndex].toString(
+	            parameterLists == null ? null : parameterLists[resultIndex]);
+    	}
```

Comment:
Let's wrap the new if-block on the outside. That way it's clear that the default of `"<unknown>"` would only be replaced if it's detailed logging is enabled:

```java
String queryString = "<unknown>";
if (pgStatement.getPGConnection().getLogServerErrorDetail()) {
    if (resultIndex < queries.length) {
        queryString = queries[resultIndex].toString(
            parameterLists == null ? null : parameterLists[resultIndex]);
      }
}
```

Reply:
👍